woocommerce/woocommerce-gutenberg-products-block
# Drop stock reservation when removing item from cart via the Store API

If a user with a draft order removes an item from their cart, this PR will free up the stock held by the order manually. Usually it would be held for 10 mins.

Caveats:

- This can only capture events from the cart page/Store API. Other store events are not captured.
- Removing an item from the cart removes the hold on the stock for the entire draft order, until they checkout again. This is probably more heavy handed than ideal (?) but we only have access to methods to release stock for the entire order, not on a cart item basis.
- This cannot react to cart item changes i.e change in qty. Again, this would be heavy handed if we dropped stock on all change events.

This does fix the case in #2843 and would lesson the impact of held stock if a user abandons checkout and edits the cart, but it has limitations. Since stock is only held for 10 minutes anyway, I think this can be considered low importance/edge case.

Fixes #2843

Note, also in this PR I removed our local copy of the ReserveStock class. Our minimum supported version of WC is 4.3 so we no longer need this.

### How to test the changes in this Pull Request:

1. Add items that are stock managed to the cart.
2. Go to checkout.
3. View the contents of your wp_wc_reserved_stock table. I use Sequel Pro.
4. Go to the cart and remove an item from the cart.
5. Confirm the stock hold is deleted from wp_wc_reserved_stock.

### Changelog

> Remove held stock for a draft order if an item is removed from the cart.
